ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -182,6 +193,23 @@ def get_table_description(
                     description = description[0]
         return description
 
+    def query_table_names_and_types(
+        self, schema_name: str
+    ) -> Iterable[TableNameAndType]:
+        """
+        Connect to the source database to get the table
+        name and type. By default, use the inspector method
+        to get the names and pass the Regular type.
+
+        This is useful for sources where we need fine-grained
+        logic on how to handle table types, e.g., external, foreign,...
+        """
+
+        return [
+            TableNameAndType(name=table_name)
+            for table_name in self.inspector.get_table_names(schema_name) or []
```

Comment:
By default, `query_table_names_and_types` calls the inspector and passes the type as `Regular` as we had before